plugin/pkg/replacer/replacer_test.go

```diff
@@ -247,6 +247,8 @@ func TestLabels(t *testing.T) {
 
 	for lbl := range labels {
 		repl := replacer.Replace(ctx, state, w, lbl)
+		// test empty state won't panic
+		replacer.Replace(ctx, request.Request{}, w, lbl)
```

Comment:
you can pass `nil` here as the 3rd parameter, to test if it doesn't panic as well